tests/jerry/array-prototype-sort.js

```diff
@@ -43,9 +43,20 @@ function f(arg1, arg2) {
 	}
 }
 
-var sorted = array.sort(f);
+array.sort(f);
 for (i = 0; i < array.length; i++) {
-	assert(sorted[sorted.length - i - 1] === i);
+	assert(array[array.length - i - 1] === i);
+}
+
+// Sorting sparse array
+var array = [1,,2,,3,,4,undefined,5];
+var expected = [1,2,3,4,5,,,,];
```

Comment:
The issue with `hasOwnProperty` is related with #319, in particular, the problem is that we currently can't distinguish between "empty" values that are unset, and explicitly defined `undefined` values, currently both are `undefined`. If we do it by the standard, then empty values should be deleted, but currently that also deletes the set `undefined` value.